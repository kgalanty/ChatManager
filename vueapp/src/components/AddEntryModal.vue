<template>
  <form action="">
    <div class="modal-card" style="width: 95vw">
      <header class="modal-card-head">
        <p class="modal-card-title">Add chat</p>
        <button type="button" class="delete" @click="$emit('close')" />
      </header>
      <section class="modal-card-body">
        <b-message type="is-info" has-icon>
          Insert chat ID. We will read it from LiveChat. You will then be able
          to review and edit all data.
        </b-message>
        <b-notification
          v-if="error"
          type="is-danger"
          has-icon
          :closable="false"
          role="alert"
        >
          {{ error }}
        </b-notification>

        <b-field label="Thread ID" :type="threadid.type"
            :message="threadid.msg">
          <b-input
            v-model="threadid.value"
            placeholder="Fill Thread ID. This cannot be edited later."
          
          ></b-input>
        </b-field>
      </section>
      <footer class="modal-card-foot">
        <b-button label="Close" @click="$emit('close')" />
        <b-button
          label="Proceed"
          type="is-primary"
          icon-right="content-save"
          @click="lookupAPI"
          :loading="proceedBtnLoading"
        />
      </footer>
    </div>
  </form>
</template>
<script>
/* eslint-disable vue/no-unused-components */
import memberMixin from "../mixins/memberMixin";
import notificationsMixin from '../mixins/notificationsMixin';
import requestsMixin from "../mixins/requestsMixin";
export default {
  name: "AddEntryModal",
  mixins: [memberMixin, requestsMixin, notificationsMixin],
  components: {},
  computed: {},
  methods: {
    emitEvent(item) {
      this.$emit("runedition", item);
    },
    resetFieldState(field)
    {
      this[field].type = ''
      this[field].msg = ''
      
    },
    showError(field, msg)
    {
      this[field].type = 'is-danger'
      this[field].msg = msg
      this[field].value = ''
    },
    lookupAPI() {
      this.error = "";
      this.resetFieldState('threadid')
      this.proceedBtnLoading = true;
      const params = this.generateParamsForRequest("LiveChat");
      this.$api
        .get(`addonmodules.php?${params}&tid=${this.threadid.value}`)
        .then((data) => {
          this.proceedBtnLoading = false;
          if (data.data.result == "success") {
            this.$emit("runedition", data.data.data);
            this.$emit("close");
          } else {

            this.showError('threadid', data.data.msg)
            //this.error = data.data.msg;
          }
        });
    },
  },
  mounted() {
    this.error = "";
  },
  data() {
    return {
      threadid:
      {
        type: '',
        msg: '',
        value: ''
      },
      error: "",
      proceedBtnLoading: false,
    };
  },
};
</script>
<style >
.reviewTabHeader {
  background: #ffcf76;
}
.modaltable {
  border: 1px solid black;
}
.modaltable thead {
  background: rgb(165, 197, 255);
  background: linear-gradient(
    180deg,
    rgba(165, 197, 255, 1) 0%,
    rgba(119, 172, 255, 1) 100%
  );
}
/* .current
{
  width: 100%;
            overflow: hidden;
            border: 2px solid #296fa8;
            border-radius: 5px;
            padding: 10px;
} */
section.tab-content {
  border-radius: 0.25rem;
  box-shadow: 0 0.5em 1em -0.125em rgb(10 10 10 / 10%),
    0 0 0 1px rgb(10 10 10 / 2%);
  color: #4a4a4a;
  max-width: 100%;
  position: relative;
}
</style>