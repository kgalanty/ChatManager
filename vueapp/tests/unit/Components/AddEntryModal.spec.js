import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import Buefy from 'buefy'
import AddEntryModal from '@/components/AddEntryModal'
import axios from 'axios'
const GlobalPlugins = {
    install(v) {
      v.prototype.$api = axios
    },
  };

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(GlobalPlugins)

//jest.mock("axios");

describe('AddEntryModal.vue', () =>
{
    const mocks = {
        $api: {
          get: Promise.resolve({data: {data: {result: 'success'}}}),
         
        },
      };
    it('renders a modal', () =>
    {
        const wrapper = shallowMount(AddEntryModal, { localVue })
        expect(wrapper).toMatchSnapshot()
    })
    it('success from api', async () =>
    {
        jest.spyOn(axios, 'get').mockResolvedValue({data: {data: {result: 'success'}}});
        const wrapper = mount(AddEntryModal, { localVue  })
        const input = wrapper.find('input')
        input.element.value = 'testvalue'
        input.trigger('input')
        const button = wrapper.find('button.is-primary')
         button.trigger('click')
       await wrapper.vm.$nextTick(()=>
       {
           console.log(wrapper.vm)
        expect(wrapper.emitted().close).toHaveBeenCalled()
        done()
       }) 
      // axios.get.mockImplementation(() => Promise.resolve({data: {data: {result: 'success'}}}));
    //    expect(wrapper.emitted().runedition).toBeTruthy()
       


    })
})