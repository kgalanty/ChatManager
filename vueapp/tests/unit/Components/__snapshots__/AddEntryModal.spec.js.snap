// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEntryModal.vue renders a modal 1`] = `
<form action="">
  <div class="modal-card" style="width: 95vw;">
    <header class="modal-card-head">
      <p class="modal-card-title">Add chat</p> <button type="button" class="delete"></button>
    </header>
    <section class="modal-card-body">
      <b-message-stub active="true" closable="true" type="is-info" hasicon="true" duration="2000">
        Insert chat ID. We will read it from LiveChat. You will then be able
        to review and edit all data.
      </b-message-stub>
      <!---->
      <b-field-stub type="" label="Thread ID" message="" addons="true">
        <b-input-stub usehtml5validation="true" statusicon="true" value="" type="text" hascounter="true" customclass="" placeholder="Fill Thread ID. This cannot be edited later."></b-input-stub>
      </b-field-stub>
    </section>
    <footer class="modal-card-foot">
      <b-button-stub label="Close" nativetype="button" tag="button"></b-button-stub>
      <b-button-stub type="is-primary" label="Proceed" iconright="content-save" nativetype="button" tag="button"></b-button-stub>
    </footer>
  </div>
</form>
`;
